Source/Charts/Data/Implementations/Standard/ChartData.swift

```diff
@@ -482,129 +324,251 @@ open class ChartData: NSObject
     /// specified index. 
     ///
     /// - Returns: `true` if an entry was removed, `false` ifno Entry was found that meets the specified requirements.
-    @objc @discardableResult open func removeEntry(xValue: Double, dataSetIndex: Int) -> Bool
+    @objc @discardableResult open func removeEntry(xValue: Double, dataSetIndex: Index) -> Bool
```

Comment:
Then this one could be `index(ofXValue:)` (on `ChartDataSet`)

Reply:
for simple improvements, please pull this branch and just update the code, usually I should be ok with it. I will check what you updated

Reply:
I suppose this and the above comment were for @danielgindi . There have been disagreements in the past around API changes related to this. We didn't have the `Collection` API then though.